src/app/admin/programs/services/query.service.ts

```diff
@@ -91,8 +91,8 @@ export class QueryService {
     }
 
     static executeParse(condition: ProgramConditionClass) : ProgramConditionClass {
-        condition.form.get('value').setValue(Number.parseInt((<string>condition.data.value), 10));
```

Comment:
condition.data.value contained the old value, so it basically just overwrote what we had in there.